src/metabase/api/table.clj

```diff
@@ -66,14 +71,18 @@
                      :description             description))
     (api/check-500 (db/update! Table id, :visibility_type visibility_type))
     (let [updated-table (Table id)
+          driver (->> id table/table-id->database-id driver/database-id->driver) ;; is this the easy way to do this?
           new-visibility (visible-state? (:visibility_type updated-table))
           old-visibility (visible-state? original-visibility-type)
           visibility-changed? (and (not= new-visibility
                                          old-visibility)
                                    (= :show new-visibility))]
       (when visibility-changed?
         (log/debug (u/format-color 'green "Table visibility changed, resyncing %s -> %s : %s") original-visibility-type visibility_type visibility-changed?)
-        (sync-database/sync-table! updated-table))
+        (sync-database/sync-table! updated-table)
+        (cached-values/cache-table-data-shape! driver updated-table)
```

Comment:
`cache-table-data-shape!`  should probably figure out the driver (if it needs it) from the `updated-table`